graphein/protein/visualisation.py

```diff
@@ -817,21 +824,61 @@ def asteroid_plot(
             node_x.append(x)
             node_y.append(y)
 
-        degrees = [
-            subgraph.degree(n) * node_size_multiplier for n in subgraph.nodes()
-        ]
+        def node_size_function(g: nx.Graph, feature: str):
```

Comment:
Do you think you could refactor this & the other similar function to reduce duplication?

https://github.com/a-r-j/graphein/blob/bb4ba769e73dfdd43097b10ba2d41f6cc6a7a73e/graphein/protein/visualisation.py#L247